### ISSUE TITLE
Switch language voice command

### ISSUE THREAD

**Issue by guit4eva** [C-Nedelcu/talk-to-chatgpt#82]
Another suggestion - which would be incredible for language learners. Have the ability to select two languages, and then a voice command to switch between them. This will allow users to quickly switch to their native language to discuss a language problem, and then switch back to the language they are learning.

**Comment by C-Nedelcu** [comment]
Yeah, I was thinking about that. Sounds good, I'll mark this as work in progress.

I also had an idea about how we could have some sort of "interpreter mode", to use ChatGPT as a real time voice translator.
What I want to achieve: you speak in language X, chatGPT translates to language Y and the response is spoken out loud in language Y.
I suppose all it would take is to be able to append something to the prompt, such as "you must reply in [language Y]".
So basically I need to add an "add to the beginning/end of prompt" option, which allows you to add whatever you want to the prompt.
* in the microphone, you say "hello my friend"
* the prompt sent to GPT is: "please translate the following to Japanese: hello my friend"
* the text-to-speech language is set Japanese
I'm excited by my own idea and I can't wait to record a youtube video of me "speaking" japanese ! 
Damn, ChatGPT is so good. So many possibilities.

**Comment by guit4eva** [comment]
That sounds like a cool idea :) Man, I'm so impressed by your extension - it is by far the best thing I have ever used in a browser!! Well done on developing it, seriously good!

**Comment by C-Nedelcu** [comment]
An additional idea to complete the "interpreter mode"

When you speak to someone in a foreign language and want ChatGPT to translate something, my initial idea works.
But what if you want a reply from the person who only speaks in a foreign language?
Then you need to switch the input/output languages back and forth, at every sentence.

I think I need to work on a full-fledged "interpreter mode" where you pick two languages, and everything is fully automated. Something like this:
* I say "hello friend" in english
* The text sent to chatGPT is : "translate the following to japanese: hello friend"
* chatGPT replies in japanese, the reply is spoken out loud in japanese
* AUTOMAGIC SWITCH - the speech recognition language switches to japanese / the prompt changes to english
* the japanese friend responds "こんにちは友達"
* the text sent to ChatGPT is: "英語に翻訳してください: こんにちは友達" (this literally means "translate to english: hello friend")
* chatGPT replies in english, the reply is spoken out loud in english

It would be as if you had an actual interpreter standing between you and your friend.

A couple of weeks from now, my mother in law will be visiting, and she only speaks mandarin, so this could be really helpful! I should develop this new feature before she visits.

**Comment by guit4eva** [comment]
That's a seriously good idea. This will have a double benefit - if you are learning a language, you could say a sentence in English, listen to the translation, and then repeat the translation and see if you were correct :) Amazing!!

Ha ha I'm sure your mother-in-law is going to be really impressed when she sees this !!! :)

**Comment by MikesMoneyMachine** [comment]
Eleven labs does this guys, it does may languages very well

**Comment by MikesMoneyMachine** [comment]
English
Spanish
French
Hindi
Italian
German
Polish
Portuguese

^^to be exact.

**Comment by bartman081523** [comment]
@C-Nedelcu 
can I ask whether this idea was successful, and does it depend wholly on Eleven Labs TTS, or does it apply also for Chrome TTS? I also think, if you add more Settings/Parameters, that it would be good if the Settings had Registers, like in Ublock Origin, f.e. But its just a suggestion, you make your own decisions for your own code, of course.

Also whishing you all good for your conversation in Mandarin in Interpreter-Mode. Hoping also to see this feature in a future version. Keep up the good work!
